apps/src/blockly/addons/cdoBlockFlyout.js

```diff
@@ -0,0 +1,212 @@
+import GoogleBlockly from 'blockly/core';
+
+export default class CdoBlockFlyout extends GoogleBlockly.HorizontalFlyout {
+  /**
+   * This is a customized flyout class that extends the HorizontalFlyout class.
+   * This flyout is intended to be place inside of a block's FieldFlyout.
+   *
+   * @param {Object} workspaceOptions - The options for constructing the class.
+   */
+  constructor(workspaceOptions) {
+    super(workspaceOptions);
+    this.horizontalLayout_ = !0;
+    this.minWidth_ = workspaceOptions.minWidth || this.minWidth_;
+    this.maxWidth_ = workspaceOptions.maxWidth || this.maxWidth_;
+  }
+
+  autoClose = false;
+  minWidth_ = 0;
+  maxWidth_ = 1000;
+
+  /**
+   * @returns True if this flyout may be scrolled with a scrollbar or
+   *     by dragging.
+   */
+  isScrollable() {
+    return false;
+  }
+
+  /**
+   * Creates the flyout's DOM.  Only needs to be called once.  The flyout can
+   * either exist as its own SVG element or be a g element nested inside a
+   * separate SVG element.
+   *
+   * @param tagName The type of tag to
+   *     put the flyout in. This should be <svg> or <g>.
+   * @returns The flyout's SVG group.
+   */
+  createDom(tagName) {
+    super.createDom(tagName);
+    tagName =
+      'flyoutClip' +
+      Blockly.utils.idGenerator.genUid().replace(/([\(\)])/g, '');
+    let definitions = Blockly.utils.dom.createSvgElement(
+      'defs',
+      {},
+      this.svgGroup_
+    );
+    definitions = Blockly.utils.dom.createSvgElement(
+      'clipPath',
+      {
+        id: tagName,
+      },
+      definitions
+    );
+    this.svgClipPath_ = Blockly.utils.dom.createSvgElement(
+      'path',
+      {},
+      definitions
+    );
+    this.svgGroup_.setAttribute('clip-path', 'url(#' + tagName + ')');
+    this.svgGroup_.classList.add('blockFieldFlyout');
+    return this.svgGroup_;
+  }
+
+  /**
+   * Initializes the flyout.
+   *
+   * @param targetWorkspace The workspace in which to
+   *     create new blocks.
+   */
+  init(targetWorkspace) {
+    super.init(targetWorkspace);
+    this.targetWorkspace_ = targetWorkspace;
+  }
+
+  /**
+   * Compute height of flyout.  Position mat under each block.
+   * For RTL: Lay out the blocks right-aligned.
+   */
+  reflowInternal_() {
+    this.height_ = 0;
+    this.width_ = 0;
+    const topBlocks = this.workspace_.getTopBlocks(false);
+    for (let i = 0, block; (block = topBlocks[i]); i++) {
```

Comment:
I haven't seen this syntax before, is there a reason we are doing this rather than a traditional for loop or a forEach over `topBlocks`?

Reply:
Oh yeah, this one is wild! This syntax comes directly from the minified code that Ben copied to get this work started. We should definitely do a quick refactor to make it more readable.

Reply:
Updated to use forEach!